### ISSUE TITLE
[Bug] Uncaught SyntaxError: Identifier has already been declared

### ISSUE THREAD

**Issue by Enteleform** [storybookjs/addon-svelte-csf#33]
### Describe the bug

Naming a story with the same name as its component results in the following error:
`Uncaught SyntaxError: Identifier 'SomeStory' has already been declared`

Related to [storybook#8536](https://github.com/storybookjs/storybook/issues/8536).

### Steps to reproduce the behavior

`./src/SomeCategory.stories.svelte`

```html
<script lang="ts">
  import SomeStory               from "./SomeStory.svelte"
  import {Meta, Template, Story} from "@storybook/addon-svelte-csf"
</script>

<Meta title="SomeCategory"/>

<Template let:args>
  <svelte:component this={args.component}/>
</Template>

<!-- issue caused by `SomeStory` being used for both the `name` attribute and import name of `args.component` -->
<Story name="SomeStory" args={{component:SomeStory}}/>
```

### Expected behavior

Storybook will render the `SomeCategory/SomeStory` story without errors.

### Environment

```
OS:      Windows 10
Node:    16.4.2
NPM:     7.18.1
Browser: Chrome 91.0.4472.164
 ```

**Comment by benbender** [comment]
@Enteleform could you provide a simplified reproduction? Without context/repro nobody will be able to reproduce/fix the issue. Generally svelte(kit) works fine with storybook.

**Comment by Enteleform** [comment]
@benbender I included a simplified reproduction in the `Steps to reproduce the behavior` section of the issue template, and just updated the example with a comment to point out the specific line that is causing the issue.

**Comment by benbender** [comment]
@Enteleform Not really. Thats a snippet, but no repro.Its missing installed/used packages, setup etc. A repro has to be complete...

**Comment by pzuraq** [comment]
Also encountering this issue, the compiled output appears to be the issue:

```js
export const __namedExportsOrder = ["Header-7f663153"];
export const Header-7f663153 = __storiesMetaData.stories["Header-7f663153"];
```

Oddly enough, it only happens to some components, others appear to work fine:

```js
export const __namedExportsOrder = ["Footer7dc3a1bb"];
export const Footer7dc3a1bb = __storiesMetaData.stories["Footer7dc3a1bb"];
```

**Comment by pzuraq** [comment]
I'm using `storybook-builder-vite` btw, in case that could be part of the issue

**Comment by pzuraq** [comment]
Fixed it by passing in an explicit story ID, should have read the warnings:

```svelte
<Story
  id="HeaderMain"
  name="Header"
  args={{
    type: 'image',
  }}
/>
```

**Comment by j3rem1e** [comment]
The method building the hashcode can returns negative hash: the generated identifier is not valid, because it contains a "-"

**Comment by shilman** [comment]
:rocket: Issue was released in [`v1.1.1`](https://github.com/storybookjs/addon-svelte-csf/releases/tag/v1.1.1) :rocket:

**Comment by ranjanified** [comment]
> Fixed it by passing in an explicit story ID, should have read the warnings:
> 
> <Story
>   id="HeaderMain"
>   name="Header"
>   args={{
>     type: 'image',
>   }}
> />

plus 1.
